### PR TITLE
10458: (1/x - 1).is_nonpositive

### PR DESCRIPTION
**Pull request by smichr**
Some minor modifications to what what has already been done plus some line-length modification of the `_monotonic_sign` function. It seems like something is not quite right in that function, but what was done (and the slight reinterpretation that I have done) is sound.

### CODE REVIEW

File: sympy/functions/elementary/tests/test_exponential.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def test_lambertw():
     assert LambertW(2, evaluate=False).is_real
     p = Symbol('p', positive=True)
     assert LambertW(p, evaluate=False).is_real
-    assert LambertW(p - 1, evaluate=False).is_real is None
+    assert LambertW(p - 1, evaluate=False).is_real is False
```
Comment (mvnnn) [inline sympy/functions/elementary/tests/test_exponential.py]: @smichr ```None``` is correct answer. ```False``` --> ```None```

Comment (smichr) [inline sympy/functions/elementary/tests/test_exponential.py]: hmm. So the change is not correct (or something else is wrong).

Comment (mvnnn) [inline sympy/functions/elementary/tests/test_exponential.py]: https://github.com/sympy/sympy/blob/master/sympy/core/add.py#L612 need to change here
i mean ```exp``` is constant not variable so we need to add in to c(constant) but still it's add with variable```a```.

Comment (mvnnn) [inline sympy/functions/elementary/tests/test_exponential.py]: if we don't do this then this type of complicated case makes problem
```
>>> p = Symbol('p', positive=True)
>>> (-p + 1/S.Exp1 - 1).is_nonnegative
```
it's return ```None``` but right ans is ```False```.many other test case are possible.

File: sympy/core/add.py [hunk 2]
```diff
@@ -594,10 +594,12 @@ def _eval_is_positive(self):
     def _eval_is_nonnegative(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
-            c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonnegative:
+            c, a = self.as_independent(*self.free_symbols)
```
Comment (mvnnn) [inline sympy/core/add.py]: problem in this ```as_independent(*self.free_symbols)``` function . that's return sometime wrong result.

Comment (mvnnn) [inline sympy/core/add.py]: @smichr problem in this ```as_independent(*self.free_symbols)``` function . that's return sometime wrong result.sometime return ```correct``` ans  for same input. try this example ::
```
>>> x = Symbol('x', integer=True, negative=True)
>>> (1 - 1/x).is_nonpositive
```
it's return ```None```
```
>>> x = Symbol('x', integer=True, negative=True)
>>> (1 - 1/x).is_nonnegative
True
>>> (1 - 1/x).is_nonpositive
False
```
that's return ```correct``` ans for same condition.

Comment (mvnnn) [inline sympy/core/add.py]: @smichr if we can improve ```as_independent()``` function then we can solve this issue. You think about another method ?

Comment (smichr) [inline sympy/core/add.py]: > if we can improve as_independent()

Can you give an example of wrong out put from the `as_independent` function?

Comment (mvnnn) [inline sympy/core/add.py]: @smichr Actually, I deleted sympy repo from my leptop and Currently, i am doing web development internship so i have no time for this. ```sorry for that```

**Comment by smichr** [comment]
stale

**Pull request closed without merging**